### PR TITLE
ARROW-17107: [Java] Fix variable-width vectors in integration JSON writer

### PR DESCRIPTION
**Pull request by lidavidm**
- Clarify that this is meant to be an internal format
- Fix handling of empty variable-width vectors

### CODE REVIEW

File: java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java [hunk 1]
```diff
@@ -83,8 +84,10 @@
 import com.fasterxml.jackson.databind.MappingJsonFactory;
 
 /**
- * A writer that converts binary Vectors into a JSON format suitable
+ * A writer that converts binary Vectors into an <em>internal, unstable</em> JSON format suitable
```
Comment (lwhite1) [inline java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java:87]: If this is intended for integration testing, could it be under the test directory?

**Review by lwhite1 — COMMENTED** [review]
One question, which I assume you've already considered. Otherwise looks good. Thank you for all the added testing.

File: java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java (hunk 1 above)
Comment (lidavidm) [reply java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileWriter.java:87]: Integration tests are run via a binary, not via JUnit, so they need to end up in something that gets compiled into a JAR. (Also I'd be a little hesitant to break compatibility immediately here.)

**Comment by lwhite1** [comment]
Makes perfect sense. Thanks

**Comment by jarohen** [comment]
Thanks @lidavidm & @lwhite1! :pray:

**Pull request merged**